packages/plugin-ext/src/main/browser/view/tree-views-main.tsx

```diff
@@ -251,6 +259,32 @@ export class TreeViewWidget extends TreeWidget {
         });
     }
 
+    protected readonly onSelectionChangedEmitter = new Emitter<SelectionEventHandler>();
+
+    get onSelectionChanged(): Event<SelectionEventHandler> {
+        return this.onSelectionChangedEmitter.event;
+    }
+
+    protected fireSelectionChanged(node: SelectableTreeNode, contextSelection: boolean): void {
+        this.onSelectionChangedEmitter.fire({node: node, contextSelection: contextSelection} as SelectionEventHandler);
+    }
+
+    protected handleClickEvent(node: TreeNode | undefined, event: React.MouseEvent<HTMLElement>): void {
+        super.handleClickEvent(node, event);
+
+        if (SelectableTreeNode.is(node)) {
+            this.fireSelectionChanged(node, false);
+        }
+    }
+
+    protected handleContextMenuEvent(node: TreeNode | undefined, event: React.MouseEvent<HTMLElement>): void {
```

Comment:
why don't override only this method and do something like:
```ts
protected _contextSelection = false;
get contextSelection(): boolean {
    return this._contextSelection;
}
protected handleContextMenuEvent(...): void {
    try {
            this._contextSelection = true;
            super.handleContextMenuEvent(node, event);
    } finally {
            this._contextSelection = false;
    }
}
```

and then `this.proxy.$setSelection(treeViewId, treeItemId, this.treeViewWidget.contextSelection);`

in this way you will make sure that any change in model selection is propagated, no matter whether it is from mouse, keyboard or done programmatically

Reply:
I've already thought about this way, but wasn't sure that this will be correct way of solving this issue. I'll try.

Reply:
Does it fix? If it does i am fine with merging, since it does not change how selection change from keyboard or other sources is handled.

Reply:
Yes, it does. At first step it will be enough.

Yesterday, I took a look on that things, how they it were implemented in file navigator and this type of tree widget (in terms of navigation and event handling). Unfortunately, there are architectural problems in tree widget, that was created during implementing plug-in API. It is not logical, that there is the only one handler that is listening to the tree modification. Modification, I mean, changing selection. Event, when we are navigating via keyboard, we send to plug-in API signals to process selection and load node content.

So, adding a fix to support correct process of navigating by keyboard won't be enough. Because, there aren't any procedures to process Enter button press (in plug-in tree implementation). At this moment this is process via single changing selection.

These modifications is not easy and has to be discussed a lot across teams, because what I see at that moment, these changes becomes growing out of the scope of this issue and may take a lot of time. The initial task was to add an ability to upload and delete Kubernetes configuration maps. 

So, what I propose.
Those fix would be enough to correct workflow, but meantime I will create a separate issue, for respecting existed behavior in tree widgets. Because this architectural changes are not in my responsibility zone.